src/Functions/fromDaysSinceYearZero.cpp

```diff
@@ -84,7 +76,8 @@ class FunctionFromDaysSinceYearZero : public IFunction
             return false;
         };
 
-        const bool success = try_type(UInt8{}) || try_type(UInt16{}) || try_type(UInt32{}) || try_type(UInt64{});
+        const bool success = try_type(UInt8{}) || try_type(UInt16{}) || try_type(UInt32{}) || try_type(UInt64{})
```

Comment:
`isNativeNumber`?

Reply:
`isNativeNumber` may be used in `getReturnTypeImpl()` but not in `executeImpl()` (at least I don't know how it could)

Reply:
Why is this doing validation in `executeImpl` and not in `getReturnTypeImpl` like most other functions?

Reply:
Another thing. Does the template need to be adjusted?

It's doing:
```
using equivalent_integer = typename std::conditional_t<sizeof(T) == 4, UInt32, UInt64>;
```

So it will use UInt64 for everything but Int32/UInt32

Reply:
@Algunenano, 
> Why is this doing validation in executeImpl and not in getReturnTypeImpl like most other functions?

So, we need to accept an Int* type cause it is the result type of subtractions in queries (like in the PR description).

We have only two options here, I believe: 
- traverse the entire column in advance in `getReturnTypeImpl` and check for negative values
- do this check within `executeImpl` and throw

> Does the template need to be adjusted?

The size of the corresponding Int/UInt types should be the same, plus we check for negatives, so I just left it as it was implemented previously.

Reply:
>  traverse the entire column in advance in getReturnTypeImpl and check for negative values

Not an option, `getReturnTypeImpl()` has no access to the actual column data, it can only see the column metadata (type, name, compression/representation).